From 12a2160868151a5dbc4a4077d02bda4a292c0d6e Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sat, 16 Nov 2024 13:10:49 +0100
Subject: [PATCH] qtwaylandscanner: don't track source location

This is a backport of upstream fix for reproducible builds, see:
https://codereview.qt-project.org/c/qt/qtwayland/+/557799

Upstream-Status: Backport [https://invent.kde.org/plasma/kwin/-/merge_requests/6789]
Signed-Off-by: Andreas Cord-Landwehr <cordlandwehr@kde.org>
---
 src/wayland/tools/qtwaylandscanner.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/wayland/tools/qtwaylandscanner.cpp b/src/wayland/tools/qtwaylandscanner.cpp
index 2e0dd03577..9c5a957599 100644
--- a/src/wayland/tools/qtwaylandscanner.cpp
+++ b/src/wayland/tools/qtwaylandscanner.cpp
@@ -39,6 +39,7 @@
 
 #include <QCoreApplication>
 #include <QFile>
+#include <QFileInfo>
 #include <QXmlStreamReader>
 
 #include <vector>
@@ -469,7 +470,7 @@ bool Scanner::process()
         return false;
 
     printf("// This file was generated by qtwaylandscanner\n");
-    printf("// source file is %s\n\n", qPrintable(m_protocolFilePath));
+    printf("// source file is %s\n\n", qPrintable(QFileInfo(file).fileName()));
 
     for (auto b : std::as_const(m_includes))
         printf("#include %s\n", b.constData());
-- 
2.43.0

